packages/components/src/components/Select/__tests__/Select.spec.tsx

```diff
@@ -0,0 +1,182 @@
+        const { result } = renderHook(() => useSelectData(menuSlot));
+        expect(result.current.filterText).toBe('');
+        expect(result.current.items.length).toBe(6);
+        act(() => {
+            result.current.setFilterText('test1');
+        });
+        expect(result.current.filterText).toBe('test1');
+        expect(result.current.items.length).toBe(1);
+    });
+});
+
+describe('getSelectOptionValue', () => {
+    it('returns correct value and label for basic item', () => {
+        const basicItem = <Select.Item value="test1">Test1</Select.Item>;
+        const item = getSelectOptionValue(basicItem.props as SelectItemProps);
+        expect(item.value).toBe('test1');
+        expect(item.label).toBe('Test1');
+    });
+
+    it('returns correct value and label for item with label', () => {
+        const basicItem = (
+            <Select.Item value="test1" label="labelvalue">
+                Test1
+            </Select.Item>
+        );
+        const item = getSelectOptionValue(basicItem.props as SelectItemProps);
+        expect(item.value).toBe('test1');
+        expect(item.label).toBe('labelvalue');
+    });
+
+    it('returns correct value and label for item with child component', () => {
+        const basicItem = (
+            <Select.Item value="test1" label="labelvalue">
+                <IconIcon />
+            </Select.Item>
+        );
+        const item = getSelectOptionValue(basicItem.props as SelectItemProps);
+        expect(item.value).toBe('test1');
+        expect(item.label).toBe('labelvalue');
+    });
+});
+
+describe('isReactLeaf', () => {
+    it('returns if item with string is a react leaf', () => {
+        const basicItem = <Select.Item value="test1">Test1</Select.Item>;
+        const isLeaf = isReactLeaf(basicItem, Select.Item);
+        expect(isLeaf).toBe(true);
+    });
+    it('returns if item with number is a react leaf', () => {
+        const basicItem = <Select.Item value="test1">123</Select.Item>;
+        const isLeaf = isReactLeaf(basicItem, Select.Item);
+        expect(isLeaf).toBe(true);
+    });
+    it('returns if group with items is a react leaf', () => {
+        const group = (
+            <Select.Group groupId="testgroup">
+                <Select.Item value="test1">Test1</Select.Item>
+                <Select.Item value="test2">Test2</Select.Item>;
+            </Select.Group>
+        );
+        const isLeaf = isReactLeaf(group, Select.Group);
+        expect(isLeaf).toBe(true);
+    });
+    it('returns if group with string is a react leaf', () => {
+        const group = <Select.Group groupId="testgroup">ABC</Select.Group>;
+        const isLeaf = isReactLeaf(group, Select.Group);
+        expect(isLeaf).toBe(true);
+    });
+    it('returns if string is a react leaf', () => {
+        const group = 'abc';
+        const isLeaf = isReactLeaf(group, Select.Item);
+        expect(isLeaf).toBe(false);
+    });
+});
+
+describe('recursiveMap', () => {
```

Comment:
Please move this `describe` to `utils.spec.tsx`

Reply:
moved